src/mo_frontend/typing.ml

```diff
@@ -55,9 +55,10 @@ type env =
     check_unused : bool;
     used_identifiers : S.t ref;
     unused_warnings : unused_warnings ref;
+    viper_mode : bool;
   }
 
-let env_of_scope msgs scope =
+let env_of_scope ?(viper_mode=false) msgs scope =
```

Comment:
I wonder if we could, instead of additing this optional argument a bunch of places, just add a field to scope and use that appropriately, to avoid changing so many function signatures.